user/api.go

```diff
@@ -539,8 +539,20 @@ func (a *Api) UpdateUser(res http.ResponseWriter, req *http.Request, vars map[st
 					failedMarketoUploadCounter.Inc()
 				}
 			}
-			a.logAudit(req, tokenData, "UpdateUser isClinic{%t}", updatedUser.IsClinic())
-			a.sendUser(res, updatedUser, tokenData.IsServer)
+			// Generate new token after update done in db (could be factorize with create User func)
+			role := "patient"
+			if updatedUser.Roles != nil && len(updatedUser.Roles) > 0 {
+				role = updatedUser.Roles[0]
+			}
+			newTokenData := token.TokenData{DurationSecs: extractTokenDuration(req), UserId: updatedUser.Id, IsServer: tokenData.IsServer, Role: role}
+			tokenConfig := token.TokenConfig{DurationSecs: a.ApiConfig.TokenDurationSecs, Secret: a.ApiConfig.Secret}
+			if sessionToken, err := CreateSessionTokenAndSave(req.Context(), &newTokenData, tokenConfig, a.Store); err != nil {
```

Comment:
This changes the behaviour of the api and it does not seems correct, updateUser should not return a new token.

Reply:
we decide to perform it here as it more consistent wit the other api /login  and "/createuser". However you were not in the call we could discuss it more of course

Reply:
As discussed, indeed the token return by the "/create user" is somehow invalid and not useable. The best solution is to modify the method _RefreshSession_ (` [GET] /login`) to return a new "correct" token.